src/Storages/MySQL/MySQLSettings.h

```diff
@@ -19,6 +19,8 @@ class ASTStorage;
     M(UInt64, connection_max_tries, 3, "Number of retries for pool with failover", 0) \
     M(UInt64, connection_wait_timeout, 5, "Timeout (in seconds) for waiting for free connection (in case of there is already connection_pool_size active connections), 0 - do not wait.", 0) \
     M(Bool, connection_auto_close, true, "Auto-close connection after query execution, i.e. disable connection reuse.", 0) \
+    M(UInt64, connect_timeout, 60, "Connect timeout", 0) \
+    M(UInt64, read_write_timeout, 1800, "Read/write timeout", 0) \
```

Comment:
Defaults does not looks sane enough.
Maybe default should be the same as DBMS_DEFAULT_RECEIVE_TIMEOUT_SEC/DBMS_DEFAULT_SEND_TIMEOUT_SEC for read_write_timeout?
And DBMS_DEFAULT_CONNECT_TIMEOUT_SEC for connect_timeout?

Reply:
Yes, seems this was is better, though I just made the defaults the same as they were before for mysql:
https://github.com/ClickHouse/ClickHouse/blob/fc4e947f4520676177cb3724cf92ad234dea1c6a/base/mysqlxx/Connection.h#L12-L13

Reply:
@azat or may be not, because `MYSQLXX_DEFAULT_TIMEOUT` is almost the same as default for `connect_timeout_with_failover_ms` and `MYSQLXX_DEFAULT_RW_TIMEOUT`is almost the same as default for `receive_data_timeout_ms`. So seems like rather sane defaults.

Reply:
Hm, maybe those defaults are also note sane, why they are so huge? 1min for connect this can cause huge query delays.

Reply:
> MYSQLXX_DEFAULT_TIMEOUT is almost the same as default for connect_timeout_with_failover_ms

bad argument, for mysql it is in seconds and `connect_timeout_with_failover_ms` is not. So I'll change to lower default.